modules/dxp/apps/analytics/analytics-reports-journal-impl/src/main/java/com/liferay/analytics/reports/journal/internal/info/item/JournalArticleAnalyticsReportsInfoItem.java

```diff
@@ -118,6 +126,77 @@ public WebImage getAuthorWebImage(
 		);
 	}
 
+	@Override
+	public String getCanonicalURL(
+		JournalArticle journalArticle, Locale locale) {
+
+		Optional<ThemeDisplay> themeDisplayOptional =
+			_getThemeDisplayOptional();
+
+		if (!themeDisplayOptional.isPresent()) {
+			return StringPool.BLANK;
+		}
+
+		return themeDisplayOptional.map(
+			themeDisplay -> {
+				Optional<Layout> layoutOptional = _getLayoutOptional(
+					journalArticle);
+
+				return layoutOptional.map(
+					layout -> {
+						HttpServletRequest httpServletRequest =
+							themeDisplay.getRequest();
+
+						LayoutDisplayPageObjectProvider
+							initialLayoutDisplayPageObjectProvider =
+								(LayoutDisplayPageObjectProvider)
+									httpServletRequest.getAttribute(
+										LayoutDisplayPageWebKeys.
+											LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER);
+
+						httpServletRequest.setAttribute(
```

Comment:
Sorry, I'm struggling with this line, why is it needed?

Reply:
I am setting the parameter LAYOUT_DISPLAY_PAGE_OBJECT_PROVIDER, so it can be used inside the LayoutSeoManager.

Reply:
You should check the code inside the LayoutSeoManager, probably things would be cleared.

Reply:
Ohh, thanks, I didn't get to `FriendlyURLMapperProvider`. Now It's clear